HiWARPs/cactus-backend
# Add endpoint to create a form and update formschema.

# What does this PR do? 

Add endpoint for the form creation feature. Forms are created within existing projects. The content is stored in mongo, using the format of the formschema.

# How to test this PR:

1. start MongoDB, e.g. using docker: ```docker run -p 27017:27017 --name mongodb -d mongo``` 
and ```export DATABASE_URL="mongodb://localhost:27017/test"```

2. start the backend server, e.g. using: ```npm start```

3. create a project, e.g.: ```curl --location 'http://localhost:3000/project' \
--header 'Content-Type: application/json' \
--data '{
    "name": "name of project"
}'```

4. create a form within an existing project, eg.: ```curl --location 'http://localhost:3000/project/6425b800328e9c670b4b27b5/form' \
--header 'Content-Type: application/json' \
--data '{ 
    "name": "form name",
    "description": "form description", 
    "references" : "form references"
}'```

5. check that the form was created by retrieving it: ```curl --location 'http://localhost:3000/project/6425b800328e9c670b4b27b5/form/6425bacdbd1388308db5a1bf'```

# Anything else the reviewer should know about?

 This PR makes the branch feature/add-forms obsolete.
